src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -650,7 +650,7 @@ private void ParseArrayMember(ParseRecord pr)
                         ParseString(pr, objectPr);
                         var = pr._value;
                     }
-                    else if (ReferenceEquals(pr._dtTypeCode, InternalPrimitiveTypeE.Invalid))
```

Comment:
Similar comment here.  This `else if` block will have _never_ executed, and now with your change it may.  If we're going to make a change here, a better one is probably to delete this `else if` block.

Reply:
Yep, sounds good. Will do that.